kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala

```diff
@@ -105,6 +105,13 @@ class KyuubiSessionImpl(
       _client = KyuubiSyncThriftClient.createClient(user, passwd, host, port, sessionConf)
       _engineSessionHandle = _client.openSession(protocol, user, passwd, optimizedConf)
       logSessionInfo(s"Connected to engine [$host:$port] with ${_engineSessionHandle}")
+
+      sessionManager.credentialsManager.sendCredentialsIfNeeded(
```

Comment:
how about moving it into `LaunchEngine`?

Reply:
Define a method in `LaunchEngine`.

Reply:
> Define a method in LaunchEngine.
So that we can add a timeout parameter for `sendCredentialsIfNeeded` operation in another pr.

Reply:
> So that we can add a timeout parameter for sendCredentialsIfNeeded operation in another pr.

The reason is that, for `HadoopFsDelegationTokenProvider`, we can config multiple fileSystem URIs.

If one of the filesystem service is stuck, it might spent long time for the `sendCredentialsIfNeeded`.